datafusion-examples/examples/custom_datasource.rs

```diff
@@ -199,11 +200,23 @@ impl CustomExec {
         db: CustomDataSource,
     ) -> Self {
         let projected_schema = project_schema(&schema, projections).unwrap();
+        let cache = Self::create_cache(projected_schema.clone());
         Self {
             db,
             projected_schema,
+            cache,
         }
     }
+
+    /// This function creates the cache object that stores the plan properties such as schema, equivalence properties, ordering, partitioning, etc.
+    fn create_cache(schema: SchemaRef) -> PlanPropertiesCache {
```

Comment:
As mentioned above, calling this like 

```suggestion
    fn compute_properties(schema: SchemaRef) -> PlanProperties {
```

I think would make its use and intent more clear